### PR TITLE
Use local FFmpeg instead of Jave's

### PR DESCRIPTION
**Pull request by rob93c**

### CODE REVIEW

**Comment by rob93c** [comment]
**Important:** the current implementation has a concurrency issue: **_n_** concurrent calls involving videos will result in **_n_** invocations of the command to retrieve FFmpeg installation path.

File: src/main/java/com/github/stickerifier/stickerify/logger/SubstringHighlighter.java [hunk 1]
```diff
@@ -22,10 +23,10 @@
 public class SubstringHighlighter extends MessageConverter {
 
 	private static final String START_YELLOW = changeColorTo(BOLD + YELLOW_FG);
-	static final String CONTINUE_PREVIOUS_COLOR = changeColorTo("0;" + DEFAULT_FG);
+	static final String CONTINUE_PREVIOUS_COLOR = changeColorTo(RESET + DEFAULT_FG);
 	static final String HIGHLIGHTED_NEW_USER = " " + START_YELLOW + NEW_USER.substring(1) + CONTINUE_PREVIOUS_COLOR;
 	static final String START_GREEN = changeColorTo(BOLD + GREEN_FG);
-	private static final Pattern MIME_TYPE_PATTERN = Pattern.compile("[\\w-]+/[\\w-]+");
+	private static final Pattern MIME_TYPE_PATTERN = Pattern.compile("( )([\\w-]+/[\\w-]+)( )");
```
Comment (MartelliEnrico) [inline src/main/java/com/github/stickerifier/stickerify/logger/SubstringHighlighter.java:29]: No need to capture the spaces around the primary capture group
```suggestion
	private static final Pattern MIME_TYPE_PATTERN = Pattern.compile(" ([\\w-]+/[\\w-]+) ");
```

File: src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java [hunk 2]
```diff
@@ -39,6 +41,7 @@ public final class MediaHelper {
 	private static final Logger LOGGER = LoggerFactory.getLogger(MediaHelper.class);
 
 	private static final Gson GSON = new Gson();
+	static final ProcessLocator FFMPEG_LOCATOR = new PathLocator();
```
Comment (MartelliEnrico) [inline src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java:44]: Maybe rename to PROCESS_LOCATOR?

File: src/main/java/com/github/stickerifier/stickerify/process/PathLocator.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(PathLocator.class);
+
+	private static final boolean IS_WINDOWS = System.getProperty("os.name").contains("Windows");
+	private static final String[] FIND_FFMPEG = { IS_WINDOWS ? "where" : "which", "ffmpeg" };
+
+	private static String ffmpegLocation = "";
+
+	public PathLocator() {
+		if (ffmpegLocation.isEmpty()) {
```
Comment (MartelliEnrico) [inline src/main/java/com/github/stickerifier/stickerify/process/PathLocator.java:23]: We should remove the check, and make ffmpegLocatation non-static

**Review by MartelliEnrico — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/stickerifier/stickerify/logger/SubstringHighlighter.java (hunk 1 above)
Comment (rob93c) [reply src/main/java/com/github/stickerifier/stickerify/logger/SubstringHighlighter.java:29]: At one point it was defined like that but I changed it mid rework, I will fix it soon 💯

File: src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java (hunk 2 above)
Comment (rob93c) [reply src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java:44]: To me it's more readable explicitly calling it "ffmpeg"

**Review by MartelliEnrico — APPROVED** [review]
LGTM

**Pull request merged**